schema/string_test.go

```diff
@@ -3,36 +3,57 @@ package schema
 import "testing"
 
 // To be in par with the python library.
-func TestCastString_URIMustRequireScheme(t *testing.T) {
-	if _, err := castString(stringURI, "google.com"); err == nil {
+func TestDecodeString_URIMustRequireScheme(t *testing.T) {
+	if _, err := decodeString(stringURI, "google.com", Constraints{}); err == nil {
 		t.Errorf("want:err got:nil")
 	}
 }
 
-func TestCastString_InvalidUUIDVersion(t *testing.T) {
+func TestDecodeString_InvalidUUIDVersion(t *testing.T) {
 	// This is a uuid3: namespace DNS and python.org.
-	if _, err := castString(stringUUID, "6fa459ea-ee8a-3ca4-894e-db77e160355e"); err == nil {
+	if _, err := decodeString(stringUUID, "6fa459ea-ee8a-3ca4-894e-db77e160355e", Constraints{}); err == nil {
 		t.Errorf("want:err got:nil")
 	}
 }
 
-func TestCastString_Success(t *testing.T) {
-	var data = []struct {
-		Desc   string
-		Value  string
-		Format string
-	}{
-		{"URI", "http://google.com", stringURI},
-		{"Email", "foo@bar.com", stringEmail},
-		{"UUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID},
-	}
-	for _, d := range data {
-		v, err := castString(d.Format, d.Value)
-		if err != nil {
-			t.Errorf("want:nil got:%q", err)
+func TestDecodeString_Success(t *testing.T) {
+	t.Run("ValidMinLength", func(t *testing.T) {
+		var data = []struct {
+			Desc        string
+			Value       string
+			Format      string
+			constraints Constraints
+		}{
+			{"URI", "http://google.com", stringURI, Constraints{MinLength: 5}},
+			{"Email", "foo@bar.com", stringEmail, Constraints{MinLength: 3}},
+			{"UUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID, Constraints{MinLength: 36}},
 		}
-		if v != d.Value {
-			t.Errorf("want:%s got:%s", d.Value, v)
+		for _, d := range data {
+			v, err := decodeString(d.Format, d.Value, d.constraints)
+			if err != nil {
+				t.Errorf("want:nil got:%q", err)
+			}
+			if v != d.Value {
+				t.Errorf("want:%s got:%s", d.Value, v)
+			}
 		}
-	}
+	})
+
+	t.Run("Error", func(t *testing.T) {
```

Comment:
It is confusing to have an "Error" subtest inside a test method which ends in Success. Could you please either rename the TestDecodeString_Success to TestDecodeString (and leave both subtests) or make TestDecodeString_error its own test function?